### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by vasyllabaruby**

### CODE REVIEW

File: lib/middlewares/racker.rb [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+module Middlewares
+  # Main controller class
+  class Racker
+    def self.call(env)
+      new(env).response.finish
+    end
+
+    def initialize(env)
+      @request = Rack::Request.new(env)
+      @game = Codebreaker::Game.new
```
Comment (nico-saj) [inline lib/middlewares/racker.rb]: Looks like u use it only for new games. Move it to `new_game`

File: lib/middlewares/racker.rb [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    def level
+      @request.session[:level]
+    end
+
+    def attempts
+      @request.session[:attempts]
+    end
+
+    def hints
+      @request.session[:hints]
```
Comment (nico-saj) [inline lib/middlewares/racker.rb]: Could u use `game.hints`?

File: lib/middlewares/racker.rb [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+    def hints_total
+      @request.session[:game].player.hints_total
+    end
+
+    def players_stats
+      @game.statistic
+    end
+
+    def number
```
Comment (nico-saj) [inline lib/middlewares/racker.rb]: This method looks weird. Method name looks like getter, but u increment number on each call

File: public/game.html.erb [hunk 4]
```diff
@@ -0,0 +1,81 @@
+<!DOCTYPE html>
+<html lang="en" class="mdl-js">
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+    <link href="assets/favicon.ico" rel="icon">
+    <link href="assets/css/bootstrap.min.css" rel="stylesheet">
+    <link href="assets/css/styles.css" rel="stylesheet">
+    <script src="assets/js/jquery.min.js"></script>
+    <script src="assets/js/bootstrap.min.js"></script>
+    <title>Codebreaker Web</title>
```
Comment (nico-saj) [inline public/game.html.erb]: Try to use I18n gem in this task.

File: lib/middlewares/racker.rb [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+      last_result = @game.play(@request.params['number'])
+      return win if last_result == '++++'
+      return lose if last_result =~ /^[1-6]{4}$/
+
+      result(last_result)
+    end
+
+    def result(result_str)
+      result_arr = result_str.chars
+      @first = result_arr[0] || 'X'
```
Comment (nico-saj) [inline lib/middlewares/racker.rb]: Use an array

File: public/statistics.html.erb [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+              <th scope="col">#</th>
+              <th scope="col">Name</th>
+              <th scope="col">Level</th>
+              <th scope="col">Attempts Left</th>
+              <th scope="col">Hints Left</th>
+              <th scope="col">Date</th>
+            </tr>
+            </thead>
+            <tbody>
+            <% players_stats.each do |player| %>
```
Comment (nico-saj) [inline public/statistics.html.erb]: Could u use `each_with_index`?

**Review by nico-saj — CHANGES_REQUESTED** [review]
Add `.ruby-version` and check tests on CircleCI

File: lib/middlewares/racker.rb (hunk 1 above)
Comment (vasyllabaruby) [reply lib/middlewares/racker.rb]: No, Codebreaker::Game used also to view statistics which can be called without starting new game

File: Gemfile.lock [hunk 7]
```diff
@@ -1,36 +0,0 @@
-GIT
```
Comment (nico-saj) [inline Gemfile.lock:1]: Why have u deleted `Gemfile.lock`?

**Review by nico-saj — CHANGES_REQUESTED** [review]
Remove node_modules from GitHub and add node_modules folder to gitignore

**Review by nico-saj — APPROVED** [review]

**Pull request merged**